packages/hardhat-foundry/src/index.ts

```diff
@@ -0,0 +1,176 @@
+import { extendConfig, internalTask, task } from "hardhat/config";
+
+import {
+  TASK_COMPILE_SOLIDITY_GET_COMPILATION_JOB_FOR_FILE,
+  TASK_COMPILE_TRANSFORM_IMPORT_NAME,
+} from "hardhat/builtin-tasks/task-names";
+import { SolidityFilesCache } from "hardhat/builtin-tasks/utils/solidity-files-cache";
+import {
+  CompilationJob,
+  CompilationJobCreationError,
+  DependencyGraph,
+  HardhatRuntimeEnvironment,
+  ResolvedFile,
+} from "hardhat/types";
+import { existsSync, writeFileSync } from "fs";
+import path from "path";
+import chalk from "chalk";
+import {
+  getForgeConfig,
+  getRemappings,
+  HardhatFoundryError,
+  installDependency,
+} from "./foundry";
+
+const TASK_INIT_FOUNDRY = "init-foundry";
+
+let pluginActivated = false;
+
+extendConfig((config, userConfig) => {
+  // Check foundry.toml presence. Don't warn when running foundry initialization task
+  if (!existsSync(path.join(config.paths.root, "foundry.toml"))) {
+    if (!process.argv.includes(TASK_INIT_FOUNDRY)) {
+      console.log(
+        chalk.yellow(
+          `Warning: You are using the hardhat-foundry plugin but there isn't a foundry.toml file in your project. Run 'npx hardhat ${TASK_INIT_FOUNDRY}' to create one.`
+        )
+      );
+    }
+    return;
+  }
+
+  // Load foundry config
+  const foundryConfig = getForgeConfig();
+
+  // Ensure required keys exist
+  if (
+    foundryConfig?.src === undefined ||
+    foundryConfig?.cache_path === undefined
+  ) {
+    throw new HardhatFoundryError(
+      "Couldn't find `src` or `cache_path` config keys after running `forge config --json`"
+    );
+  }
+
+  // Ensure foundry src path doesnt mismatch user-configured path
+  const userSourcesPath = userConfig.paths?.sources;
+  const foundrySourcesPath = foundryConfig.src;
+
+  if (userSourcesPath !== undefined && userSourcesPath !== foundrySourcesPath) {
+    throw new HardhatFoundryError(
+      `User-configured sources path (${userSourcesPath}) doesn't match path configured in foundry (${foundrySourcesPath})`
```

Comment:
I think this error needs a better explanation. At least a hint about what the users should do.

Reply:
@alcuadrado what about:

```
The sources path in your Hardhat config (...) doesn't match
the path configured in Foundry (...).

Make sure that they use the same path, or remove the custom
sources path in your Hardhat config to let the plugin handle it for you.
```